### PR TITLE
Make plot_api return all observations for response

### PR DESCRIPTION
**Pull request by yngve-sk**
**Issue**
Resolves [#9077](https://github.com/equinor/ert/issues/9077)

Tested on RGS for stable and bleeding:
Before (left is rendering only one instead of 3 observations):

After:
Screenshot 2024-10-30 at 11 56 24

### CODE REVIEW

File: src/ert/gui/tools/plot/plot_api.py [hunk 1]
```diff
@@ -205,29 +205,37 @@ def observations_for_key(self, ensemble_ids: List[str], key: str) -> pd.DataFram
                     timeout=self._timeout,
                 )
                 self._check_response(response)
+
+                observations = response.json()
+                observations_dfs = []
+
                 if not response.json():
```
Comment (DanSava) [inline src/ert/gui/tools/plot/plot_api.py:212]: Why not replace `if not response.json():` with `if not observations:` if you already declared `observations = response.json()` on ln 209

File: src/ert/gui/tools/plot/plot_api.py [hunk 2]
```diff
@@ -205,29 +205,37 @@ def observations_for_key(self, ensemble_ids: List[str], key: str) -> pd.DataFram
...
                     continue
                 try:
-                    obs = response.json()[0]
+                    response.json()[0]
```
Comment (DanSava) [inline src/ert/gui/tools/plot/plot_api.py:215]: Similar why not replace `response.json()[0]` with `observations[0]`

**Review by frode-aarstad — APPROVED** [review]
LGTM

File: src/ert/gui/tools/plot/plot_api.py [hunk 3]
```diff
@@ -205,29 +205,37 @@ def observations_for_key(self, ensemble_ids: List[str], key: str) -> pd.DataFram
...
                 except (KeyError, IndexError, JSONDecodeError) as e:
                     raise httpx.RequestError(
                         f"Observation schema might have changed key={key},  ensemble_name={ensemble.name}, e={e}"
                     ) from e
```
Comment (DanSava) [inline src/ert/gui/tools/plot/plot_api.py:219]: It is unclear to me what the purpose of this section is now.

**Review by DanSava — APPROVED** [review]
Looks ok as is but with some structure/style comments :racehorse:

File: src/ert/gui/tools/plot/plot_api.py (hunk 3 above)
Comment (yngve-sk) [reply src/ert/gui/tools/plot/plot_api.py:219]: I'm not sure either, but am also not confident to just remove it here as part of this bugfix, although I think this entire module, PlotAPI/Dark storage could be simplified alot

Comment (DanSava) [reply src/ert/gui/tools/plot/plot_api.py]: I agree with keeping it for now

**Pull request merged**